### ISSUE TITLE
Fan out input

### ISSUE THREAD

**Issue by patrobinson** [Jeffail/benthos#144]
We have an existing tool that processes individual messages against a list of "rules". If any rule matches the resulting payload is then enriched/transformed and forwarded to a sink.

I would like to replicate this with Benthos, but the only way I can seem to do that is combining all rules into a complex or/and filter. This would make them difficult to modify and reason about.

It would be preferable to "fan out" inputs to multiple processors. Further, it would be useful to define a DAG that allows all processors to forward to a single enrichment/transformation processor.

```
          -> rule 1 --
         /              \
input --  -> rule 2 --    --> enrich --> transform --> output
         \              /
          -> rule 3 --
```

Let me know your thoughts

**Comment by Jeffail** [comment]
Hey @patrobinson, for complex filters and flows we use the [`process_dag`](https://github.com/Jeffail/benthos/tree/master/docs/processors#process_dag) processor, which automatically resolves a DAG against a map of [`process_map`](https://github.com/Jeffail/benthos/tree/master/docs/processors#process_map) processors and executes them in parallel tiers.

If you can break your needs down into an example I'm more than happy to try and fit it into this model, then we can see where it falls short and discuss potential solutions.

**Comment by Jeffail** [comment]
Hey @patrobinson, I've put together a document describing workflows in Benthos in more detail: https://github.com/Jeffail/benthos/blob/master/docs/workflows.md, if you find this works for you that's great, but otherwise it would be interesting to hear more about your use case.

**Comment by Jeffail** [comment]
Hey @patrobinson, closing this for now but feel free to reopen if you have questions or feedback.

**Comment by patrobinson** [comment]
Yeah sorry, it's not a priority right now. Will get back to you at a later date
